xml/System/UriBuilder.xml

```diff
@@ -627,17 +627,23 @@
       </ReturnValue>
       <Docs>
         <summary>Gets or sets the fragment portion of the URI.</summary>
-        <value>The fragment portion of the URI. The fragment identifier ("#") is added to the beginning of the fragment.</value>
+        <value>The fragment portion of the URI.</value>
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
-## Remarks  
- The <xref:System.UriBuilder.Fragment%2A> property contains any text following a fragment marker (#) in the URI, including the marker itself. When setting the <xref:System.UriBuilder.Fragment%2A> property, the fragment marker is added to its value. **Note** In version 1.0 and 1.1 of the .NET Framework, setting the <xref:System.UriBuilder.Fragment%2A> property to any value, including `null`, cleared the <xref:System.UriBuilder.Query%2A> property, and setting the <xref:System.UriBuilder.Fragment%2A> property to `null` or to <xref:System.String.Empty?displayProperty=nameWithType> cleared the property. In version 2.0, however, the <xref:System.UriBuilder.Fragment%2A> and <xref:System.UriBuilder.Query%2A> properties are independent.  
-  
- **Note** Do not append a string directly to this property. If the length of <xref:System.UriBuilder.Fragment%2A> is greater than 1, retrieve the property value as a string, remove the leading pound sign, append the new query string, and set the property with the combined string.  
-  
-   
-  
+## Remarks
+
+The <xref:System.UriBuilder.Fragment> property contains any text following a fragment marker (#) in the URI, including the marker itself. When setting the <xref:System.UriBuilder.Fragment> property:
+
+- In .NET Framework, the fragment marker is always prepended to the string.
+- In .NET Core and later versions, the fragment marker is prepended to the string if it's not already present.
+
+> [!NOTE]
+> In version 1.0 and 1.1 of .NET Framework, setting the <xref:System.UriBuilder.Fragment> property to any value, including `null`, cleared the <xref:System.UriBuilder.Query> property, and setting the <xref:System.UriBuilder.Fragment> property to `null` or to <xref:System.String.Empty?displayProperty=nameWithType> cleared the property. In .NET Framework 2.0 and later versions, however, the <xref:System.UriBuilder.Fragment> and <xref:System.UriBuilder.Query> properties are independent.
+
+> [!NOTE]
+> When appending a string to this property, there's a behavior difference between .NET Framework and later versions of .NET to be aware of. In .NET (including .NET Core), you can simply retrieve the property, append the new string, and then set the property. However, in .NET Framework, if the length of <xref:System.UriBuilder.Fragment> is greater than 1, you must remove the leading fragment marker before appending the new fragment string and setting the property.
```

Comment:
```suggestion
> To append a value in .NET 5 (and .NET Core) and later versions, you can retrieve the property value, append the new string, and then set the property. In .NET Framework, if the length of <xref:System.UriBuilder.Fragment> is greater than 1, you must remove the leading fragment marker before appending the new fragment string and setting the property.
```

Reply:
```suggestion
> To append a value to an existing fragment in .NET Framework, you must remove the leading fragment marker before setting the property with the new value. This is because .NET Framework always prepends the fragment marker when setting the property .NET 5 (and .NET Core) and later versions are tolerant to a leading fragment marker, and will only prepend one if necessary.
```